pkg/cdi/cache.go

```diff
@@ -529,17 +530,24 @@ func (w *watch) watch(fsw *fsnotify.Watcher, m *sync.Mutex, refresh func() error
 	if watch == nil {
 		return
 	}
+
+	eventMask := fsnotify.Rename | fsnotify.Remove | fsnotify.Write
```

Comment:
@bart0sh after your probing I spent some more time investigating this, and noted that the events that are generated on `darwin` are `CREATE` events and not `WRITE` events.

I thought it safer to not modify existing behaviour on non-darwin platforms.

Reply:
Thanks for investigation. I'm wondering why CREATE was not considered on Linux. I believe it's also generated when file is created. Do we intentionally skip it?

Reply:
@klihub since you were the one who implemented this, maybe you could comment.

Note that `fsnotify` uses two different sources for the events on Linux and MacOS systems. It may also be that the `os.Rename` calls that are use in the tests trigger different events.

Reply:
> Thanks for investigation. I'm wondering why CREATE was not considered on Linux. I believe it's also generated when file is created. Do we intentionally skip it?

On Linux,
  - when you create a non-empty file you always get an fsnotify.Write event (in addition to the create)
  - when you rename a file, you always get at least an fsnotify.Rename event (plus a create if the renamed file is in the same directory)
Therefore, it should not be necessary to trigger a rescan for a fsnotify.Create.

Reply:
>  I spent some more time investigating this, and noted that the events that are generated on darwin are CREATE events and not WRITE events

Nice.

>  two different sources for the events on Linux and MacOS systems. 

Trivia. macOS itself provides two different mechanisms: FSEvents, kqueue. fsnotify seems to use kqueue.

Linux has its own mechanism, called [inotify](https://linux.die.net/man/7/inotify). [Here](https://news.ycombinator.com/item?id=8669235) is a bikeshed thread comparing both interfaces.

I like the docs of https://pythonhosted.org/watchdog/installation.html#supported-platforms-and-caveats.

Interesting: https://watchexec.github.io/docs/macos-fsevents.html